Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Cancel.hs

```diff
@@ -290,11 +290,18 @@ driverDistanceToPickup booking merchantOperatingCityId tripStartPos tripEndPos =
 
 getCancellationDuesDetails :: (Id Person.Person, Id Merchant.Merchant) -> Flow CancellationDuesDetailsRes
 getCancellationDuesDetails (personId, merchantId) = do
-  person <- QP.findById personId >>= fromMaybeM (PersonNotFound personId.getId) >>= decrypt
-  merchant <- CQM.findById merchantId >>= fromMaybeM (MerchantNotFound merchantId.getId)
-  -- TODO (Rupak): Add the logic to get the cancellation dues from fare parameters instead of BPP
-  case (person.mobileNumber, person.mobileCountryCode) of
-    (Just mobileNumber, Just countryCode) -> do
-      res <- CallBPPInternal.getCancellationDuesDetails merchant.driverOfferApiKey merchant.driverOfferBaseUrl merchant.driverOfferMerchantId mobileNumber countryCode person.currentCity
-      return $ CancellationDuesDetailsRes {cancellationDues = res.customerCancellationDuesWithCurrency, disputeChancesUsed = Just res.disputeChancesUsed, canBlockCustomer = res.canBlockCustomer}
-    _ -> throwError (PersonMobileNumberIsNULL person.id.getId)
+  booking <- QRB.findAssignedByRiderId personId
+  ride <- maybe (return Nothing) (QR.findActiveByRBId . (.id)) booking
+  let cancellationFees = (.cancellationFeeIfCancelled) =<< ride
+      currency = (.estimatedFare.currency) <$> booking
+  case (cancellationFees, currency) of
+    (Just customerCancellationDues, Just bookingCurrency) -> do
+      return $ CancellationDuesDetailsRes {cancellationDues = Just PriceAPIEntity {amount = customerCancellationDues, currency = bookingCurrency}, disputeChancesUsed = Nothing, canBlockCustomer = Nothing}
+    _ -> do
+      person <- QP.findById personId >>= fromMaybeM (PersonNotFound personId.getId) >>= decrypt
```

Comment:
Why this extra logic of calling the BPP?

Reply:
If we are polling this API in case of Bridge then according to the logic it'll have cancellationFee in ride table.
We'll return it directly without making any BPP call.
If it is not there then we can make call to BPP to fetch customer cancellation dues from rider details table at BPP side.

Reply:
Can we store it at BAP side only in person table, the way we are storing in ride table. This way we don't need to store in RiderDetails at BPP side.